custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomMainActivity.kt

```diff
@@ -19,25 +19,83 @@
 package org.kiwix.kiwixmobile.custom.main
 
 import android.os.Bundle
+import android.view.MenuItem
+import androidx.appcompat.widget.Toolbar
+import androidx.core.os.bundleOf
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.navigation.NavController
+import androidx.navigation.findNavController
+import com.google.android.material.navigation.NavigationView
+import kotlinx.android.synthetic.main.activity_main.custom_drawer_container
+import kotlinx.android.synthetic.main.activity_main.drawer_nav_view
 import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.intent
 import org.kiwix.kiwixmobile.core.main.CoreMainActivity
+import org.kiwix.kiwixmobile.core.main.ZIM_FILE_URI_KEY
+import org.kiwix.kiwixmobile.core.utils.REQUEST_PREFERENCES
 import org.kiwix.kiwixmobile.custom.R
 import org.kiwix.kiwixmobile.custom.customActivityComponent
+import org.kiwix.kiwixmobile.custom.settings.CustomSettingsActivity
 
 const val REQUEST_READ_FOR_OBB = 5002
 
 class CustomMainActivity : CoreMainActivity() {
 
+  override val navController: NavController by lazy {
+    findNavController(R.id.custom_nav_controller)
+  }
+  override val drawerContainerLayout: DrawerLayout by lazy { custom_drawer_container }
+  override val drawerNavView: NavigationView by lazy { drawer_nav_view }
+  override val bookmarksFragmentResId: Int = R.id.bookmarksFragment
+  override val historyFragmentResId: Int = R.id.historyFragment
+  override val cachedComponent by lazy { customActivityComponent }
+  override val topLevelDestinations =
+    setOf(R.id.customReaderFragment)
+
   override fun injection(coreComponent: CoreComponent) {
     customActivityComponent.inject(this)
   }
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
     if (savedInstanceState != null) {
       return
     }
-    supportFragmentManager.beginTransaction()
-      .add(R.id.fragment_custom_app_container, CustomReaderFragment()).commit()
+  }
+
+  override fun onPostCreate(savedInstanceState: Bundle?) {
+    super.onPostCreate(savedInstanceState)
```

Comment:
postCreate, also this code looks like it fits in Core

Reply:
This would have to be called after the creation lifecycle step in Core? I'd agree with this answer https://stackoverflow.com/a/16564121 that it's ok to use in a parent activity.

Reply:
![image](https://user-images.githubusercontent.com/3358282/90246191-f1ce2380-de2b-11ea-91b3-881de05d6bee.png)

So call it in onStart if it happens after `OnCreate` or just at the end of OnCreate. I truly hate this method as `postCreate` is not what happens immediately after `onCreate`

Reply:
Oof, it's a lie

Reply:
Hmm or I guess everything that happens after oncreate could be called postOnCreate

Reply:
yeah but it violates the principle of Least Surprise. I think the typical assumption is `this happens immediately after create` and you have to read the docs to discover it is a bastard.

Every android developer has to know the activity lifecycle, if only for it being asked in every interview, so lean on that shared knowledge